test/controllers/api/base_controller_subclass_test.rb

```diff
@@ -29,15 +30,38 @@ def nested_values
   def authorized
     true
   end
+
+  def action_permission
+    case params[:action]
+    when 'nested_values', 'required_nested_values', 'optional_nested_values', 'raise_error', 'authorized'
+      :view
+    else
+      super
+    end
+  end
 end
 
 class Testable < ApplicationRecord
+  include Authorizable
   belongs_to :domain
+  belongs_to :foo
+end
+
+class Foo < ApplicationRecord
+  has_many :testables
 end
 
 class Api::TestableControllerTest < ActionController::TestCase
   tests Api::TestableController
 
+  ActiveRecord::Migration.create_table(:testables) { |t| t.integer :foo_id }
+  ActiveRecord::Migration.create_table(:foos)
+
+  Minitest.after_run do
+    ActiveRecord::Migration.drop_table(:testables)
```

Comment:
It seems to break the plugin tests: http://ci.theforeman.org/job/test_plugin_pull_request/4558/#showFailuresLink .

Reply:
I'm working on it - my guess is that it breaks tests because after 1 rake task, this gets called & tests needing this table won't be able to run

Reply:
https://github.com/theforeman/foreman/pull/5226